src/dftbp/dftb/mdftb.F90

```diff
@@ -0,0 +1,1575 @@
+!--------------------------------------------------------------------------------------------------!
+!  DFTB+: general package for performing fast atomistic simulations                                !
+!  Copyright (C) 2006 - 2024  DFTB+ developers group                                               !
+!                                                                                                  !
+!  See the LICENSE file for terms of usage and distribution.                                       !
+!--------------------------------------------------------------------------------------------------!
+
+#:include 'common.fypp'
+#:include 'error.fypp'
+
+!> Routines implementing the (One-center approximation) multipole expansion for the 2nd order DFTB.
+module dftbp_dftb_mdftb
+  use dftbp_io_message, only : error
+  use dftbp_math_matrixops, only : adjointLowerTriangle
+  use dftbp_common_accuracy, only : dp, lc
+  use dftbp_common_environment, only : TEnvironment
+  use dftbp_common_globalenv, only : stdOut
+  use dftbp_type_commontypes, only : TOrbitals
+  use dftbp_type_multipole, only : TMultipole
+  use dftbp_dftb_nonscc, only : TNonSccDiff
+  use dftbp_dftb_slakocont, only : TSlakoCont
+  use dftbp_dftb_periodic, only : TNeighbourList, getNrOfNeighbours
+  use dftbp_dftb_shortgammafuncs, only : expGammaPrime, expGammaDoublePrime, expGammaTriplePrime,&
+      & expGammaQuadruplePrime, expGammaQuintuplePrime
```

Comment:
Please put module imports in alphabetical order by name of the modules in the `use` directives...